### ISSUE TITLE
Lots of miscellaneous definitions are being created in swagger, how can I limit these to a package?

### ISSUE THREAD

**Issue by ahatzz11** [springfox/springfox#2497]
- Version: 2.9.0

I'm currently running into an odd issue where I am getting a lot of miscellaneous models being added to my definitions. Is there a way to control which packages Springfox looks at for definition creation?

Here is my current configuration, which lives in a `@Component` that has `@EnableSwagger2`: 

```    
@Bean
Docket api() {
    return new Docket(DocumentationType.SWAGGER_2)
            .select()
            .apis(RequestHandlerSelectors.basePackage("com.my.package.controller"))
            .build()
            .pathMapping('/')
}
```

and here is a small subset of the definitions being generated:

```"definitions": {
    "ASTNode": {
      "type": "object",
      "properties": {
        "columnNumber": {
          "type": "integer",
          "format": "int32"
        },
        "lastColumnNumber": {
          "type": "integer",
          "format": "int32"
        },
        "lastLineNumber": {
          "type": "integer",
          "format": "int32"
        },
        "lineNumber": {
          "type": "integer",
          "format": "int32"
        },
        "nodeMetaData": {
          "type": "object"
        },
        "sourcePosition": {
          "$ref": "#/definitions/ASTNode"
        },
        "text": {
          "type": "string"
        }
      },
      "title": "ASTNode"
    },
    "AnnotatedType": {
      "type": "object",
      "properties": {
        "annotations": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Annotation"
          }
        },
        "declaredAnnotations": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Annotation"
          }
        },
        "type": {
          "$ref": "#/definitions/Type"
        }
      },
      "title": "AnnotatedType"
    },
    "Annotation": {
      "type": "object",
      "title": "Annotation"
    },
    ```

**Comment by dilipkrish** [comment]
You could mark the types you dont want to see as ignored types. However its interesting that you're seeing these types. So it seems like one of your types has dependent schemas that generate models for these types

**Comment by ahatzz11** [comment]
The types that are added are not models that exist in my code. Some seem to be groovy models. There are probably close to 100, and my application only has 4 or 5. Is there a way to only look at a certain package for models, similar to how you can limit the packages used for endpoints through `.apis()`?

**Comment by ahatzz11** [comment]
I just deleted all of my models except for one to test this out. This leftover model has a single `String` in it. When I generate my swagger through Springfox it is still pulling in all of these additional models. Here is the model `TestRequest.groovy`:

```
package com.my.package.hey

import com.fasterxml.jackson.annotation.JsonProperty
import groovy.transform.Canonical

@Canonical
class TestRequest {

    @JsonProperty('test_prop')
    String testProp
}

```

I tried to convert this model to a java class - `TestRequest.java`, and all of the extra models in the swagger spec disappeared. It seems there's an issue with between groovy and springfox. 

Some additional details:
```
groovyVersion = '2.4.15'
springBootVersion = '2.0.2.RELEASE'
springfoxVersion ='2.9.0'
swaggerVersion = '1.5.10'
```

---
I continued to play and I think I found a solution. I was looking at #752 and decided to try ignoring the groovy `MetaClass` in my docket creation.

```
.ignoredParameterTypes(MetaClass)
```

This ends up removing all of the extra classes that were created in the definitions.

Is this intended behavior? It would be cool if springfox was able to automatically filter out all the groovy classes that are found. This creates a lot of noise in swagger specs.
